src/cswinrt/code_writers.h

```diff
@@ -9037,11 +9074,33 @@ private static unsafe int Do_Abi_Invoke%
             type.TypeName(),
             type_params,
             [&](writer& w) {
-                if (type_params.empty()) return;
-                w.write(R"(
-public static Guid PIID = GuidGenerator.CreateIID(typeof(%));)",
-                    type_name
-                );
+                if (!type_params.empty())
+                {
+                    // Generating both PIID and IID for backcompat consistency and to have a common property to rely on
+                    // similar to what we do in the manual projections.
+                    w.write(R"(
+public static global::System.Guid PIID = GuidGenerator.CreateIID(typeof(%));
+public static global::System.Guid IID => PIID;
+)",
+                        type_name);
+                }
+                else
+                {
+                    if (settings.netstandard_compat)
+                    {
+                        w.write(R"(
+public static global::System.Guid IID{ get; } = new Guid(new byte[]{ % });
+)",
+                            bind<write_guid_bytes>(type));
+                    }
+                    else
+                    {
+                        w.write(R"(
+public static global::System.Guid IID { get; } = new Guid(new global::System.ReadOnlySpan<byte>(new byte[] { % }));
```

Comment:
Why not generate the same code as in out T4 file for GUIDs, to directly return RVA fields as IIDs?

Reply:
Yea I think that makes sense, updated both places where we generate similar IID to be in the same format.